avro-builder/src/main/java/com/linkedin/avroutil1/builder/operations/codegen/own/AvroUtilCodeGenOp.java

```diff
@@ -100,6 +100,10 @@ public void run(OperationContext opContext) throws Exception {
 
     writeJavaFilesToDisk(specificRecords, config.getOutputSpecificRecordClassesRoot());
 
+    Set<File> allAvroFiles = new HashSet<>(avscFiles);
+    allAvroFiles.addAll(nonImportableFiles);
+    opContext.setAvroFiles(allAvroFiles);
```

Comment:
make the collection unmodifiable for extra safety? dont give plugin authors the ability to assume this is mutable (in a meaningful way)

Reply:
How do you recommend doing that? I see Guava has an `ImmutableSet` class but I was wondering if you had a better way in mind

Reply:
Collections.unmodifiable...() family of methods. built into t he JDK